### PR TITLE
Add optional VPN settings

### PR DESCRIPTION
**Pull request by lardbit**
This branches off #66 and uses docker's extend feature to reduce configuration duplication.

Some caveats were that you can't extend a service that uses the `depends_on` key so I had to create a couple levels of base services to get around that.  

Bring up the vpn version:

    docker-compose -f docker-compose.transmission-vpn.yml up -d

### CODE REVIEW

**Comment by voidrot** [comment]
everything worked on the vpn compose, was able to start all the services and VPN using the .env file and the vpn compose, configured jackett, and added a movie to the watch list and it was successfully downloaded and working.

**Comment by lardbit** [comment]
Thanks for the contribution @voidrot!

File: env.template [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
 NUM_CELERY_WORKERS=0
+
+#########################################################
+# VPN Settings - leave blank if not using a vpn
+# https://haugene.github.io/docker-transmission-openvpn/
+#########################################################
+OPENVPN_PROVIDER=
+OPENVPN_USERNAME=
+OPENVPN_PASSWORD=
+OPENVPN_CONFIG=
```
Comment (panmau) [inline env.template:19]: This env variable is not exposed to the vpn docker container per default. I think it definitely should be!

Comment (lardbit) [reply env.template:19]: whoops! You're right.  I swear I tested it so I'm at a loss as to how it was working.  I'll add it.

Comment (panmau) [reply env.template:19]: I created a PR: #116

**Pull request merged**